graaljs/graaljs-webpack-guide/README.md

```diff
@@ -1,31 +1,121 @@
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [1. Setting Up the Java Project](#1-setting-up-the-java-project)
+  - [Create a new Gradle Java Project](#create-a-new-gradle-java-project)
+  - [Create a new Maven Java Project](#create-a-new-maven-java-project)
+- [2. Setting Up the JavaScript Build](#2-setting-up-the-javascript-build)
+- [3. Using the JavaScript Library from Java](#3-using-the-javascript-library-from-java)
+- [4. Building and Running the Application](#4-building-and-running-the-application)
+- [Conclusion](#conclusion)
 
+## Introduction
+
+JavaScript libraries can be integrated into standard Java applications using [GraalJS](https://www.graalvm.org/javascript), which enables Java to execute JavaScript code.
+You can do this with the [GraalVM Polyglot API](https://www.graalvm.org/latest/reference-manual/embed-languages/), which allows you to embed and interact with JavaScript directly from Java code.
+
+GraalJS is available as both [Gradle](https://www.graalvm.org/javascript) and [Maven](https://central.sonatype.com/artifact/org.graalvm.polyglot/js) artifacts.
+This makes it easy to add to your project regardless of your build tool.
 Using Node (NPM) packages in Java projects often requires a bit more setup, due to the nature of the Node packaging ecosystem.
 One way to use such modules is to prepackage them into a single _.js_ or _.mjs_ file using a bundler like [webpack](https://webpack.js.org/).
-This guide explains step-by-step how to integrate the `webpack` build into a Maven Java project and embed the generated JavaScript code in the JAR file of the application.
 
-# GraalJS QRCode Demo
+In this guide, you will integrate the `webpack` build into your Java project and embed the generated JavaScript code in your application's JAR file.
 
-## 1. Getting Started
+## Prerequisites
 
 In this guide, you will add the [qrcode](https://www.npmjs.com/package/qrcode) NPM package to a Java application to generate QR codes.
 
-To complete this guide, you need the following:
+To do this you will need:
+
+- JDK 21 or later
+- Gradle 8.0 or later, or Maven 3.6.3 or later
+- A decent text editor or IDE
+
+---
+
+## 1. Setting Up the Java Project
+
+You can use either **Gradle** or **Maven** to set up your Java project:
+
+- for Gradle projects, navigate to the Gradle folder with `cd gradle`
+- for Maven projects, navigate to the Maven folder with `cd maven`
+
+### Create a new Gradle Java Project
+
+You can start with any Gradle Java project.
+
+If you don’t have one yet run the following:
+
+```shell
+gradle init --type java-application
+```
+
+#### Add the GraalJS Dependencies
+
+Add the following required GraalJS dependencies to the `dependencies` block of your `build.gradle` file:
```

Comment:
It's not critical but we have this rule to italicize the name of files and paths:
```
...your _build.gradle_ file:
```